ddtrace/opentelemetry/tracer_provider.go

```diff
@@ -0,0 +1,84 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+package opentelemetry
+
+import (
+	"sync"
+	"sync/atomic"
+	"time"
+
+	oteltrace "go.opentelemetry.io/otel/trace"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/internal"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+)
+
+var _ oteltrace.TracerProvider = (*TracerProvider)(nil)
+
+// TracerProvider provides implementation of OpenTelemetry TracerProvider interface.
+// TracerProvider provides Tracers that are used by instrumentation code to
+// trace computational workflows.
+// WithInstrumentationVersion and WithSchemaURL TracerOptions are not supported.
+type TracerProvider struct {
+	tracer  *oteltracer
+	ddopts  []tracer.StartOption
+	stopped atomic.Bool
+	sync.Once
+}
+
+// NewTracerProvider returns an instance of OpenTelemetry TracerProvider with Datadog Tracer start options.
+// This allows to propagate parameters to tracer.Start function.
+func NewTracerProvider(opts ...tracer.StartOption) *TracerProvider {
+	return &TracerProvider{ddopts: opts}
+}
+
+// Tracer returns an instance of OpenTelemetry Tracer and initializes Datadog Tracer.
```

Comment:
```suggestion
// Tracer returns an instance of OpenTelemetry Tracer and initializes Datadog Tracer.
// If the TracerProvider has already been shut down, this will return a no-op tracer.
```